### DISCUSSION TITLE
How to create USB key from Finnix iso ?

### DISCUSSION THREAD

**Discussion by ABelliqueux** [finnix/finnix#42] [Q&A]
EDIT: Using `dd` like so works fine:

```
sudo dd if=finnix-12X.iso of=/dev/sdX bs=4M && sync
```

____

I'm trying to create a bootable Finnix USB disk from the latest iso.

So the website says :
> Write it to a USB flash drive [...]

But after browsing the doc, issues and Discussion sections, I have yet to find any information about how I'm supposed to do that.

I've tried using `dd` with :  

```
sudo dd if=finnix-124.iso of=/dev/sda bs=1M oflag=dsync status=progress && sync
```

and when trying to boot from the USB key, I get an error message:

```
XZ-compressed data is corrupt
 -- System halted
```

I've tried with 123 and 124 isos, checked sha256 sums, etc. 
As there seems to be no documentation about creating the boot media from the iso file, I can't be positive wether my issue is due to a bad methodology or something else.

Can someone shed some light on the correct way USB media is supposed to be created ?

**Comment by rfinnie** [comment]
Yep, that `dd` looks perfectly fine; there's no documentation because it should just be as easy as `dd`ing it to the drive or using a standard user-friendly USB-writing frontend, so the issue of documentation has never come up. :worried:  

Do you get that error before or after the logo boot screen? It kind of sounds like an issue with the USB key itself, or maybe a memory issue. Do any other live images (Debian, Ubuntu, etc) work on the same hardware?

**Comment by ABelliqueux** [comment]
Hi, thanks for your reply.

I get this error right after hitting enter in the Grub menu.

I've switched to a manjaro image on the same device and it worked fine with the same 'dd' command.

FWIW, the device is a 8GB `0951:1666 Kingston Technology DataTraveler 100 G3/G4/SE9 G2/50`, advertised as 'DataTraveler 3.0' by `fdisk -l`. 

I know I had success booting Finnix 123 from this very device a few months back, but I since had to reflash it for various task and only today did I encounter this issue. From my `.bash_history`, I see I used a slightly different dd command back then :

```
sudo dd if=finnix-124.iso of=/dev/sda bs=4M && sync
```

Trying to reflash with this command **did result in a working key,** so it might be the other dd options that are the culprit.

About the doc, I suggest to update it with at least some suggestions as to how this should be done ( mentionning image writer, dd, unetbootin, etcher, etc. ), especially since the most proeminent web result when searching the web for "create finnix usb key" is this :  https://rmprepusb.com/tutorials/011-make-an-image-recovery-usb-drive-using-finnix/  
which seems convoluted for the least, and might even be completely wrong.
